addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanClassicDeviceHandler.java

```diff
@@ -118,18 +120,26 @@ public void channelUnlinked(@NonNull ChannelUID channelUID) {
     }
 
     protected boolean addListener(Channel channel) {
-        if (channel != null && channel.getChannelTypeUID().getId().startsWith("rockerswitchListener")) {
+        if (channel == null) {
+            return true;
+        }
+
+        ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+        String id = channelTypeUID == null ? "" : channelTypeUID.getId();
+
+        if (id.startsWith("rockerswitchListener")) {
             EnOceanChannelRockerSwitchListenerConfig config = channel.getConfiguration()
                     .as(EnOceanChannelRockerSwitchListenerConfig.class);
 
             try {
-                Long enoceanId = Long.parseLong(config.enoceanId, 16);
+                Long.parseLong(config.enoceanId, 16);
                 channelConfigById.put(channel.getUID().getId(), config);
                 currentEnOceanId = config.enoceanId;
                 getBridgeHandler().addPacketListener(this);
 
                 return true;
             } catch (Exception e) {
+
```

Comment:
Just some general remarks about logging and exceptions:
1. Don't catch `Exception`. Either the specific checked exceptions and/or `RuntimeException`.
1. Out current policy is that logging to error is reserved for system wide failures. So bindings have should maximum log at warn level.
1. Here is an empty exception. Always log either to debug or trace. You may never know what problem might occur in the field and if it's caused here you have now way of knowing that it happens.